metadata/src/main/java/org/apache/kafka/image/S3StreamsMetadataImage.java

```diff
@@ -626,7 +636,7 @@ static class GetObjectsContext {
         LocalStreamRangeIndexCache indexCache;
 
         CompletableFuture<InRangeObjects> cf = new CompletableFuture<>();
-        Map<Long, Optional<StreamOffsetRange>> object2range = new HashMap<>();
+        Map<Long, StreamOffsetRange> object2range = new ConcurrentHashMap<>();
```

Comment:
change to ConcurrentHashMap to protect multi-thread access.
although the current objectStorage read-callback-thread is single-thread.